rts/lra/jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -271,6 +271,28 @@ public void filter(ContainerRequestContext containerRequestContext) {
                 lraId = incommingLRA;
         }
 
+        String query = incommingLRA == null ? null : incommingLRA.getQuery();
+
+        if (query != null) {
+            for (String param : query.split("&")) { // or apply a regex matching parent=...
+                if (param.startsWith("parent=")) {
+                    String parent = param.split("=", 2)[1];
+
+                    try {
+                        // verify that parent corresponds to a valid URI
+                        new URI(parent);
+                        // the parent param was generated using URI.toASCIIString
+                        headers.putSingle(LRA_HTTP_PARENT_CONTEXT_HEADER, parent);
```

Comment:
this will override user-provided header. Is it intended behavior?

Reply:
Come on, read the spec, aren't we allowed to implement it. Does user behaviour override the spec now (ie does the users' opinion of what constitutes the transaction hierarchy trump actuality).

Reply:
https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java#L82

-- I think we should fail if the user propagates the wrong parent header, or at least log warning. This seems like it can hide potential issues with user code.

Reply:
That part of the spec is referring to LRA_HTTP_CONTEXT_HEADER.
You can't randomly change the parent context since it forms a strict hierarchy.